pkg/frontend/admin_supportedvmsizes_list_test.go

```diff
@@ -0,0 +1,74 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	"github.com/go-test/deep"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/validate"
+)
+
+func TestSupportedvmsizes(t *testing.T) {
+	mastervmsizes := validate.SupportedVMSizesByRole(validate.VMRoleMaster)
+	workervmsizes := validate.SupportedVMSizesByRole(validate.VMRoleWorker)
+
+	type test struct {
+		name         string
+		vmRole       string
+		wantResponse *map[api.VMSize]api.VMSizeStruct
+		wantError    string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:         "vmRole is invalid",
+			vmRole:       "invalidVMRole",
+			wantError:    `400: InvalidParameter: : The provided vmRole 'invalidVMRole' is invalid. vmRole can only be master or worker`,
+			wantResponse: nil,
+		},
+		{
+			name:         "vmRole is empty",
+			vmRole:       "",
+			wantError:    `400: InvalidParameter: : The provided vmRole '' is invalid. vmRole can only be master or worker`,
+			wantResponse: nil,
+		},
+		{
+			name:         "master as vmRole",
+			vmRole:       "master",
+			wantError:    "",
+			wantResponse: &mastervmsizes,
+		},
+		{
+			name:         "worker as vmRole",
+			vmRole:       "worker",
+			wantError:    "",
+			wantResponse: &workervmsizes,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			f := &frontend{}
+			gotResponse, err := f.supportedVMSizesForRole(tt.vmRole)
+			if err != nil && err.Error() != tt.wantError || err == nil && tt.wantError != "" {
+				t.Error(err)
+			}
+			// if gotResponse and wantResponse are nil, skip
+			if gotResponse == nil && tt.wantResponse == nil {
+				t.Skip()
+			}
+			v := reflect.New(reflect.TypeOf(tt.wantResponse).Elem()).Interface()
```

Comment:
`reflect` seems like overkill here when we already know this type is going to be `*map[api.VMSize]api.VMSizeStruct`... Is there a strong reason we need to use `reflect` here, or can we just unmarshal the json into a strong type we expect? I'd rather avoid `reflect`'s use at all costs, since it loses us all the benefits of using a strongly typed language.